cabfile/cabfile.go

```diff
@@ -117,9 +121,59 @@ func New(r io.ReadSeeker) (*Cabinet, error) {
 
 	// CFHEADER
 	var hdr cfHeader
-	if err := binary.Read(r, binary.LittleEndian, &hdr); err != nil {
-		return nil, fmt.Errorf("could not deserialize header: %v", err)
+	if err := binary.Read(r, binary.LittleEndian, &hdr.Signature); err != nil {
```

Comment:
I briefly wondered if there's a more convenient way (alike encoding/json), but it looks like the answer is no, unrolling is just the straightforward way to handle this.